src/Ravenfire/BoardGameGeek/BoardGameGeekSource.php

```diff
@@ -153,21 +119,87 @@ public function saveGameInfo(string $gameId, $client)
             if ($this->gameKeyChecker($responseBoardgame, $essentialColumn) and is_array($responseBoardgame[$essentialColumn])) {
                 $responseBoardgame[$essentialColumn] = $responseBoardgame[$essentialColumn][0];
             } else if (!$this->gameKeyChecker($responseBoardgame, $essentialColumn)) {
+                $this->notice("Essential game information missing. Skipping to next game.");
                 return false;
             }
         }
 
+        $saveNeeded = false;
         $gameKey = GameModel::gameKey($responseBoardgame['name'], $responseBoardgame['boardgamepublisher'], $responseBoardgame['yearpublished']);
         $game = GameModel::where('game_key', $gameKey)->get();
+
         if (count($game) === 0) {
             $game = new GameModel();
             $game->game_key = $gameKey;
-        } else {
-            $this->notice("Game already exists. Skipping to next game");
+            $this->createNewGame($response, $game);
+            $saveNeeded = true;
+        } else{
+            $this->alert("Game already exists. Checking for updates");
+        }
+
+        $bggColumns = [
+            "playingtime" => "average_playingtime",
+            "age" => "for_player_ages",
+            "boardgamemechanic" => "boardgame_mechanic",
+            "thumbnail" => "thumbnail",
+            "image" => "image",
+            "boardgamefamily" => "boardgame_family",
+            "boardgamecategory" => "boardgame_category",
+            "boardgamedesigner" => "boardgame_designer",
+            "boardgameversion" => "boardgame_version",
+            "comment" => "comments"
+        ];
+
+        $gameId = explode("?", $gameId)[0];
+        $boardgamegeekgame = BoardGameGeekModel::where('bgg_foreign_id', $gameId)->get();
+        $boardgamegeek = new BoardGameGeekModel();
+
+        if(!$saveNeeded){
+            $boardgamegeek->game_id = $game[0]->id;
+            $game = null;
+        };
+
+        $boardgamegeek->bgg_foreign_id = $gameId;
+        $this->createNewBoardGameGeek($response, $bggColumns, $boardgamegeek);
+
+        if(count($boardgamegeekgame) !== 0){
+            foreach($bggColumns as $bggColumn){
+                if($boardgamegeekgame[0]->$bggColumn != $boardgamegeek->$bggColumn){
+                    dd($boardgamegeekgame[0]->$bggColumn, $boardgamegeek->$bggColumn);
```

Comment:
I know this is a draft. Just a friendly reminder not to forget about this `dd()`